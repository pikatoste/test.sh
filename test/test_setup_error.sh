#!/bin/bash
source "$(dirname "$(readlink -f "$0")")"/../test.sh

OUT=$TEST_SCRIPT_DIR/.test_setup_error.sh.out
FILES_DIR=$TEST_SCRIPT_DIR/files

start_test "When setup_test_suite fails, an error message should be displayed in the main output"
! INLINE=     run_test_script "$TEST_SCRIPT_DIR"/do_test_error_reporting.sh [setup_test_suite] >"$OUT" 2>&1 || false
diff "$FILES_DIR"/test_setup_error.1.out "$OUT"
rm -f "$OUT"
! INLINE=true run_test_script "$TEST_SCRIPT_DIR"/do_test_error_reporting.sh [setup_test_suite] >"$OUT" 2>&1 || false
diff "$FILES_DIR"/test_setup_error.1.out "$OUT"
rm -f "$OUT"

start_test "When setup_test fails a failed test error message should be displayed in the main output"
! INLINE=     run_test_script "$TEST_SCRIPT_DIR"/do_test_error_reporting.sh [setup_test] >"$OUT" 2>&1 || false
diff "$FILES_DIR"/test_setup_error.2.out "$OUT"
rm -f "$OUT"
! INLINE=true run_test_script "$TEST_SCRIPT_DIR"/do_test_error_reporting.sh [setup_test] >"$OUT" 2>&1 || false
diff "$FILES_DIR"/test_setup_error.3.out "$OUT"
rm -f "$OUT"
