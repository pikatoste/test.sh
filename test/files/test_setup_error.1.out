[0;31m[ERROR] setup_test_suite failed, see do_test_error_reporting.sh.out for more information[0m
